tests/conftest.py

```diff
@@ -0,0 +1,61 @@
+# Fixtures for doctests
+from pathlib import Path
+import pytest
+from time import perf_counter
+from datetime import timedelta
+import os
+import sys
+from pytest_pretty import CustomTerminalReporter
+
+from _pytest.pathlib import bestrelpath
+
+
+@pytest.fixture(autouse=True)
+def handle_np2():
+    # TODO: remove when we require numpy >= 2
+    try:
+        import numpy as np
+
+        np.set_printoptions(legacy="1.21")
+    except ImportError:
+        pass
+
+
+class SKTerminalReporter(CustomTerminalReporter):
+    """
+    This ia s custom terminal reporter to how long it takes to finish given part of tests.
+    It prints time each time when test from different file is started.
+
+    It is created to be able to see if timeout is caused by long time execution, or it is just hanging.
+    """
+
+    currentfspath: Path | None
+    _start_time: float | None
+
+    def write_fspath_result(self, nodeid: str, res, **markup: bool) -> None:
+        if getattr(self, '_start_time', None) is None:
+            self._start_time = perf_counter()
+        fspath = self.config.rootpath / nodeid.split('::')[0]
+        if self.currentfspath is None or fspath != self.currentfspath:
+            if self.currentfspath is not None and self._show_progress_info:
+                self._write_progress_information_filling_space()
```

Comment:
This is internal private function of superclass? If so, perhaps a comment with link to the original, or a comment to explain what it does. Wasn't sure if this printed to stdout, because next line does a `self.write`.

Reply:
added